File: 基礎題庫/Contents/e621/e621.py
import sys

for n in sys.stdin:
    n = int(n)

    for i in range(n):
        a, b, c = map(int, input().split())

        space = 0
        if c == 0:
            print("No free parking spaces.")
        else:
            a += 1
            while a % c != 0 and a < b:
                print(str(a) + " ", end="")
                space = 1
                a += 1
            
            gap = 1
            while a + gap < b:
                print(str(a + gap) + " ", end="")
                space = 1
                gap += 1

                if gap == c:
                    gap = 1
                    a += c
            
            if space == 0:
                print("No free parking spaces.", end="")
            print()

File: 基礎題庫/Contents/e621/e621.c
#include<stdio.h>

int main(void)
{
	int n, a, b, c;
	
	while(scanf("%d", &n) != EOF)
	{
		for(int i = 0; i < n; i++)
		{
			scanf("%d %d %d", &a, &b, &c);
			
			int space = 0;
			if(c == 1)
				printf("No free parking spaces.\n");
			else
			{
				a++;
				while(a % c != 0 && a < b)
				{
					printf("%d ", a);
					space = 1;
					a++;
				}
				
				int gap = 1;
				while(a + gap < b)
				{
					printf("%d ", a + gap);
					space = 1;
					gap++;
					
					if(gap == c)
					{
						gap = 1;
						a += c;	
					}
				}
				
				if(!space)
					printf("No free parking spaces.");
				printf("\n");
			}
		}
	}
	
	return 0;
}
